arch/x86/core/fatal.c

```diff
@@ -90,7 +90,15 @@ bool z_x86_check_stack_bounds(uintptr_t addr, size_t size, uint16_t cs)
 					_current->stack_info.size);
 	}
 
-	return (addr <= start) || (addr + size > end);
+	bool check_stack;
+
+	if ((addr <= start) || (addr + size > end)) {
```

Comment:
This line introduces a MISRA rule 12.1 violation:

```
Missing explicit parentheses on sub-expression: "addr + size".
```
You can fix it by changing the line to: 

```
if ((addr <= start) || ((addr + size) > end)) {
```

There are a lot of 12.1 violations already in the codebase, so this isn't a huge issue, but I noticed it after I used an internal scanning tool so I thought I should mention it at least. :+1: